File: BinaryTree/php/TreeNode.php
<?php

declare(strict_types=1);

if (!class_exists('TreeNode')) {
    class TreeNode
    {
        public int $val;
        public ?TreeNode $left;
        public ?TreeNode $right;

        public function __construct(int $val, ?TreeNode $left = null, ?TreeNode $right = null)
        {
            $this->val = $val;
            $this->left = $left;
            $this->right = $right;
        }
    }
}

File: BinaryTree/csharp/TreeNode.cs
namespace BinaryTreeSolutions;

public class TreeNode
{
    public int Val { get; }
    public TreeNode? Left { get; set; }
    public TreeNode? Right { get; set; }

    public TreeNode(int val, TreeNode? left = null, TreeNode? right = null)
    {
        Val = val;
        Left = left;
        Right = right;
    }
}
